### PR TITLE
Rename variables and types in query-scheduler RequestQueue

### PR DESCRIPTION
**Pull request by charleskorn**
#### What this PR does

This PR clarifies the names of some variables and types used in the query-scheduler's `RequestQueue`.

#### Which issue(s) this PR fixes or relates to

(none)

#### Checklist

- [n/a] Tests updated
- [n/a] Documentation added
- [n/a] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/scheduler/queue/queue.go
Comment (pstibrany) [inline pkg/scheduler/queue/queue.go:392]: `getNextRequestForQuerierCall` sounds like method name, not name of type to me. I'd suggest to rename it to a noun that describes the type. Verbs are used for methods.

File: pkg/scheduler/queue/queue.go
Comment (francoposa) [inline pkg/scheduler/queue/queue.go:399]: element is never read, do we need it at all

File: pkg/scheduler/queue/queue.go
Comment (francoposa) [reply pkg/scheduler/queue/queue.go:392]: some possibilities:
`readyQuerier`
`readyQuerierConnection`/`readyQuerierConn`
`waitingQuerierConnection`/`waitingQuerierConn`

File: pkg/scheduler/queue/queue.go
Comment (francoposa) [inline pkg/scheduler/queue/queue.go:113]: similar naming here once we settle on the type naming

File: pkg/scheduler/queue/queue.go
Comment (francoposa) [inline pkg/scheduler/queue/queue.go:112]: `enqueueRequest`/`enqueueRequests` are similarly verbs

the channel could be `requestsToEnqueue` or `requestsToDispatch`

the type could just be a private `request` type or `schedulerRequest` or `tenantRequest` or `queryRequest`, or `schedulerDispatchRequest`, or even `schedulerEnqueueRequests`, which still has the verb but at least clarifies which queue we are talking about

File: pkg/scheduler/queue/queue.go
Comment (charleskorn) [reply pkg/scheduler/queue/queue.go:399]: Good catch, that was left over from a previous approach to removing requests when their context was cancelled. I've removed it.

File: pkg/scheduler/queue/queue.go
Comment (charleskorn) [reply pkg/scheduler/queue/queue.go:392]: I don't love the name either... I wanted to make it clear that this represented a call to the `GetNextRequestForQuerier()` method that is waiting for a query.

I intentionally avoided names like `querier` or `querierConnection` as those are different concepts - a querier has many connections, and a single connection (managed by `Scheduler.QuerierLoop()`) calls `GetNextRequestForQuerier()` many times during its life.

File: pkg/scheduler/queue/queue.go
Comment (charleskorn) [reply pkg/scheduler/queue/queue.go:112]: I like `requestsToEnqueue`, thanks for the suggestion. I'll change it now.

File: pkg/scheduler/queue/queue.go
Comment (francoposa) [reply pkg/scheduler/queue/queue.go:392]: I don’t feel strongly enough to hold this up, I think we’ve definitely improved with the naming changes in the PR.

As some of the changes settle we could always rename later if something better makes itself apparent

Comment (francoposa) [reply pkg/scheduler/queue/queue.go:392]: maybe just dropping the `get` is enough non-verbiness for now

Comment (charleskorn) [reply pkg/scheduler/queue/queue.go:392]: That would work - might make it `waitingNextRequestForQuerierCall` to make it super clear.

**Review by francoposa — APPROVED** [review]
I'm fine to roll with this after the merge conflict is fixed.

If I were to bikeshed over one last name I'd change this setup

`nextRequestForQuerierCalls chan *waitingNextRequestForQuerierCall`

to just

`nextRequestForQuerierCalls chan *nextRequestForQuerierCall`, as the type itself doesn't necessarily indicate waiting (it could be being processed already or done), the fact that they're in a channel implies well enough that they're waiting to me.

**Review by pstibrany — APPROVED** [review]
🙈 I'm not going to start another round of naming comments 😄, approving to unblock you.

**Review by aknuds1 — APPROVED** [review]

**Pull request merged**
